sycl/test-e2e/lit.cfg.py

```diff
@@ -36,18 +39,33 @@
 # To be filled by lit.local.cfg files.
 config.required_features = []
 config.unsupported_features = []
+# Get the set of all features possible in E2E tests.
+all_features = get_all_sycl_lit_features()
+
+# Please do not call config.available_features.add directly. Use this function.
+# This wrapper function checks if the feature is contained in all_features
+# and updates available_features. It ensures all new available features are added
+# to all_features.
+def register_feature(config, all_features, feature_name, condition):
+    if feature_name not in all_features:
+        raise Exception(
+            feature_name
+            + " is not presented in all_features. Please update the set in sycl/test-e2e/sycl_lit_features.py."
+        )
+    if condition:
+        config.available_features.add(feature_name)
 
 # test-mode: Set if tests should run normally or only build/run
 config.test_mode = lit_config.params.get("test-mode", "full")
 config.fallback_build_run_only = False
 if config.test_mode == "full":
-    config.available_features.add("run-mode")
-    config.available_features.add("build-and-run-mode")
+    register_feature(config, all_features, "run-mode", True)
+    register_feature(config, all_features, "build-and-run-mode", True)
 elif config.test_mode == "run-only":
     lit_config.note("run-only test mode enabled, only executing tests")
-    config.available_features.add("run-mode")
+    register_feature(config, all_features, "run-mode", True)
```

Comment:
One more thing here: let's say we dropped this line with `register_feature("run-mode")`. The feature itself is still listed in our known features - therefore it can be used to silently disable tests forever, right?

I suppose that extra changes should be made (and maintained) to make sure that `register_feature` is always called unconditionally.

Then at the end of `lit.cfg.py` we should compare list of registered features with list of known features.
If some known feature wasn't registered at all, then its a bug and we should highlight it. By "registered" here I mean passed to `register_feature` regardless of the last argument